packages/istanbul-lib-report/lib/file-writer.js

```diff
@@ -2,193 +2,196 @@
  * @class ContentWriter
  * @constructor
  */
-/* istanbul ignore next: abstract class */
-function ContentWriter() {}
-
-/**
- * writes a string as-is to the destination
- * @param {String} str the string to write
- */
-/* istanbul ignore next: abstract class */
-ContentWriter.prototype.write = function() {
-    throw new Error('write: must be overridden');
-};
+class ContentWriter {
+    /**
+     * writes a string as-is to the destination
+     * @param {String} str the string to write
+     */
+    /* istanbul ignore next: abstract class */
+    write() {
+        throw new Error('write: must be overridden');
+    }
 
-/**
- * returns the colorized version of a string. Typically,
- * content writers that write to files will return the
- * same string and ones writing to a tty will wrap it in
- * appropriate escape sequences.
- * @param {String} str the string to colorize
- * @param {String} clazz one of `high`, `medium` or `low`
- * @returns {String} the colorized form of the string
- */
-ContentWriter.prototype.colorize = function(str /*, clazz*/) {
-    return str;
-};
+    /**
+     * returns the colorized version of a string. Typically,
+     * content writers that write to files will return the
+     * same string and ones writing to a tty will wrap it in
+     * appropriate escape sequences.
+     * @param {String} str the string to colorize
+     * @param {String} clazz one of `high`, `medium` or `low`
+     * @returns {String} the colorized form of the string
+     */
+    colorize(str /*, clazz*/) {
+        return str;
+    }
 
-/**
- * writes a string appended with a newline to the destination
- * @param {String} str the string to write
- */
-ContentWriter.prototype.println = function(str) {
-    this.write(str + '\n');
-};
+    /**
+     * writes a string appended with a newline to the destination
+     * @param {String} str the string to write
+     */
+    println(str) {
+        this.write(str + '\n');
+    }
 
-/**
- * closes this content writer. Should be called after all writes are complete.
- */
-ContentWriter.prototype.close = function() {};
+    /**
+     * closes this content writer. Should be called after all writes are complete.
+     */
+    close() {}
+}
 
 /**
  * a content writer that writes to a file
  * @param {Number} fd - the file descriptor
  * @extends ContentWriter
  * @constructor
  */
-function FileContentWriter(fd) {
-    this.fd = fd;
-}
-util.inherits(FileContentWriter, ContentWriter);
+class FileContentWriter extends ContentWriter {
+    constructor(fd) {
+        super();
+
+        this.fd = fd;
+    }
 
-FileContentWriter.prototype.write = function(str) {
-    fs.writeSync(this.fd, str);
-};
+    write(str) {
+        fs.writeSync(this.fd, str);
+    }
+
+    close() {
+        fs.closeSync(this.fd);
+    }
+}
 
-FileContentWriter.prototype.close = function() {
-    fs.closeSync(this.fd);
-};
+// allow stdout to be captured for tests.
+let capture = false;
+let output = '';
```

Comment:
I don't really understand this change, would you mind explaining it? This looks like "state" to me and shouldn't that be in the [base] class?

Reply:
This is not a "change", just moved code.

Really IMO this `capture` functionality should be removed, the test should just replace `process.stdout`.  I'd rather avoid adding complexity to this PR with a change like this.